bundles/org.openhab.binding.sonyprojector/src/main/java/org/openhab/binding/sonyprojector/internal/communication/SonyProjectorConnector.java

```diff
@@ -881,6 +882,63 @@ public void disableXvColor() throws SonyProjectorException {
         setSetting(SonyProjectorItem.XVCOLOR, XVCOLOR_OFF);
     }
 
+    /**
+     * Send an IR command to the projector
+     *
+     * @param command the IR command
+     *
+     * @throws SonyProjectorException in case this IR command is not available or any other problem
+     */
+    public void sendIrCommand(String command) throws SonyProjectorException {
+        byte[] irCode = new byte[] { 0x00, 0x00 };
+        boolean found = true;
+        if (command.startsWith("INPUT_")) {
+            try {
+                irCode = model.getInputFromName(command.substring(6)).getIrCode();
+            } catch (SonyProjectorException e) {
+                found = false;
+            }
+        } else if (command.startsWith("PRESET_")) {
+            try {
+                irCode = model.getCalibrPresetFromName(command.substring(7)).getIrCode();
+            } catch (SonyProjectorException e) {
+                found = false;
+            }
+        } else if (command.startsWith("ASPECT_")) {
+            try {
+                irCode = model.getAspectFromName(command.substring(7)).getIrCode();
+            } catch (SonyProjectorException e) {
+                found = false;
+            }
+        } else {
+            try {
+                irCode = SonyProjectorItem.getFromValue(command).getIrCode();
+            } catch (SonyProjectorException e) {
+                found = false;
+            }
+        }
+        if (!found) {
```

Comment:
I'm not sure if it's relevant or applicable here but worth noting.  It looks like Sony has both a 15-bit and 20-bit command structure that they accept for IR codes (plus possibly a 32-bit code).  It looks like this is just accepting the 15-bit.  Is there  any benefit to permitting the 20-bit codes?

Reply:
Yes, it is already supported, this is the first byte value that identifies that.

Reply:
Would the (cmd.length() != 4) not cause an issue with the 20-bit?  I would think they would need to be length of 5.

Reply:
No, it is always 2 bytes with the first byte being either 0x17 or 0x19 or 0x1B.
That is not clear for me why they call category 15 bit and 20 bit.